include/libopencm3/usb/usbstd.h

```diff
@@ -163,7 +163,8 @@ struct usb_config_descriptor {
 
 	/* Descriptor ends here.  The following are used internally: */
 	const struct usb_interface {
-		int num_altsetting;
+		uint8_t * cur_altsetting;
+		uint8_t num_altsetting;
```

Comment:
as per usb specs, bInterfaceNumber ,bAlternateSetting, bNumInterfaces are 8bit wide and unsigned (as seen in descriptor).
any benifit of increasing the width (or change sign)? afaif, No.
also, added benefit ram saving :p